src/api/ledger/parse/orderbook-order.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+const _ = require('lodash');
+const utils = require('./utils');
+const flags = utils.core.Remote.flags.offer;
+const parseAmount = require('./amount');
+
+function parseOrderbookOrder(order: Object): Object {
+  const direction = (order.Flags & flags.Sell) === 0 ? 'buy' : 'sell';
+  const takerGetsAmount = parseAmount(order.TakerGets);
+  const takerPaysAmount = parseAmount(order.TakerPays);
+  const quantity = (direction === 'buy') ? takerPaysAmount : takerGetsAmount;
+  const totalPrice = (direction === 'buy') ? takerGetsAmount : takerPaysAmount;
+
+  const specification = utils.removeUndefined({
+    direction: direction,
+    quantity: quantity,
+    totalPrice: totalPrice,
+    passive: ((order.Flags & flags.Passive) !== 0) || undefined
+  });
+  // "quality" is omitted intentionally as it corresponds to
```

Comment:
>>> Semantically the request should only return "half" of the results.

You've changed the usual semantics, giving more weight it seems to the Sell
flag to determine buy/sell - > bid/ask.
Does it really matter to a trader whether someone is semantically buying
USD or selling XRP ( for USD) ?
I guess yes, if only to get a truer sense of how much output is available
in each offer (which is dependent on the execution rate)

Hrmmm.

Was there any discussion about this?
Is it actually a useful change, and a useful distinction to make?
This is not something I'm really well suited to judge.

>>> I'm not sure if it's any more correct to say that orders get executed
at the rate of the "quality" than it is to say they get executed at the
ratio specified by "TakerGets/TakerPays"

You have it the wrong way around. The `quality` is actually `TakerPays
divided by TakerGets` and means how much you must pay for 1 unit of
TakerGets.

And it's more correct, in that the `quality` field in the json, is derived
from the quality baked into the directory nodes where the offer is indexed.
Offers are always executed at this rate, determined at the time of
placement, rather than recalculating the ratio each time. With partial
executions, the ratio may slip.

>>> "price" would be an approximation.

As would totalPrice

>>> they can use either one safely

Personally, I'd rather use quality when it's oriented correctly. i.e. where
possible.